### ISSUE TITLE
Outline for an individual object doesn't work properly

### ISSUE THREAD

**Issue by XtroTheArctic** [Chishikii/URP-Render-Features#1]
Hello. I am moving the conversation from the issue we were talking in Robinseibold's repo to here.
https://github.com/Robinseibold/Unity-URP-Outlines/issues/15#issuecomment-1903664960

In the other issue, you mentioned that this outline effect now must use "full screen" shader. I'm not a graphics developer and I don't know about scriptable render pipeline but this sounds wrong to me somehow.

If a feature wants to render an effect on a per-object basis, why would it be forced to use a full screen shader by the new Unity version (or the new URP version).

If we look at how the screenspacenormals texture was rendered in the old version (Robinseibold's outline + Unity 2022), we can clearly see that the object (my build area pre-baked mesh) was being rendered as a single, isolated object. It should not be affected or blocked by any other object in the scene to be able to render its individual outline.
![image](https://github.com/Chishikii/URP-Render-Features/assets/6631429/36ce99f6-9417-4891-8a0d-62e7f863a60a)

Somehow we need to find a way to make this shader unlit again; instead of fullscreen.

Since I am not experienced in this area much, can you please try to explain me why it has to be a fullscreen shader in the new Unity?

**Comment by Chishikii** [comment]
Have you checked after my last commit? 

We're already doing exacta that in the first phase of the pass. Drawing the renderers to a separate texture. 

Fullscreen shader is just another word for post processing which is just another word for blotting one texture to another while applying a shader. SSAO for example. That's exactly what the original outline shader did. Units just updated their api to enable us to make these post processing effects easier in URP. Ironically it's a lot easier in HDRP or built in but alas. 

So we're ;

1. Collecting all renderers we want to draw based on drawing settings and filter settings (InitRendererList)

2. Rendering these objects

3. Taking these rendered objects applying out outlines, rendering them into a temporary buffer (1. Blit)

4 rendering that back to the camera (2. Blit)

This als happens in the execute function.

**Comment by XtroTheArctic** [comment]
Oh, I didn't notice you made an update. I tried it now and... OMG!! You fixed it!

Thank you very much!!!

![image](https://github.com/Chishikii/URP-Render-Features/assets/6631429/5f631c76-12da-459e-b6a4-f88040d871b3)

**Comment by Chishikii** [comment]
I don't quite think that it's a proper fix yet, as depth testing won't work properly. 

It's more of a work around and I've noticed post processing in general isn't quite working yet either. 

If you encounter any more issues let me know. I'll add the occlusion feature sometime the next few days.

**Comment by XtroTheArctic** [comment]
Yea. I created another issue for screen edge bleeding.

**Comment by adrian-miasik** [comment]
Ayyee very nice! Glad it's rendering properly now!

So looking over it, the RTHandle system can't save color and depth data at the same time? To workaround this depth issue, you provided the list of renderers in back to front order using the [SortingCriteria](https://docs.unity3d.com/ScriptReference/Rendering.SortingCriteria.html). 

Damn, that's clever.

It seems like another alternative solution would be to create another RTHandle just for depth?
